photon-lib/src/main/scala/com/linkedin/photon/ml/util/ClassUtils.scala

```diff
@@ -22,15 +22,15 @@ object ClassUtils {
   private val ANON_CLASS_MARKER = "$anon$"
 
   /**
-   * Method for detecting anonymous classes (the isAnonymousClass method is broken in Scala).
+   * Method for detecting anonymous classes (the isAnonymousClass method is broken in Scala as late as 2.11).
    *
    * @param clazz A class
    * @return True if the given class is anonymous, false otherwise.
    */
   def isAnonClass(clazz: Class[_]): Boolean = clazz.getName.contains(ANON_CLASS_MARKER)
```

Comment:
It's only used as you describe above - I found that I needed to add the code to make up for the broken `isAnonymousClass` method. The reason I needed that method is due to the way modelTypes are copied currently (so that they're not recomputed over and over again).a

Coincidentally, I spent half a day trying to get `modelType` copied without anonymous classes and it's just a mess when dealing with the `RandomEffectModel` thanks to how Scala handles constructors. I've given up.